core-rust/state-manager/src/transaction/series_execution.rs

```diff
@@ -123,26 +123,32 @@ where
     pub fn execute(
         &mut self,
         transaction: &ValidatedLedgerTransaction,
-        description: impl Display,
+        description: &'static str,
     ) -> Result<ProcessedCommitResult, RejectResult> {
         let description = DescribedTransactionHash {
             ledger_hash: transaction.ledger_transaction_hash(),
             description,
         };
-        let config_type = transaction.config_type();
+        self.execute_wrapped(
+            &description,
+            self.execution_configurator
+                .wrap_ledger_transaction(transaction, &description),
+        )
+    }
+
+    fn execute_wrapped<T: for<'l> TransactionLogic<StagedStore<'l, S>>>(
+        &mut self,
+        description: &DescribedTransactionHash,
+        wrapped_executable: T,
+    ) -> Result<ProcessedCommitResult, RejectResult> {
         let mut execution_cache = self.execution_cache.lock();
         let processed = execution_cache.execute_transaction(
             self.store,
             self.epoch_identifiers(),
             self.state_tracker.state_version,
             &self.state_tracker.ledger_hashes.transaction_root,
-            &transaction.ledger_transaction_hash(),
-            self.execution_configurator
-                .wrap(transaction.get_executable(), config_type)
-                .warn_after(
-                    config_type.get_transaction_runtime_warn_threshold(),
-                    &description,
-                ),
+            &description.ledger_hash,
```

Comment:
hmm, this argument drives logic, and I tremble in fear when I see that we take its value from a log-only error-surfacing util struct O_O

is there a reason to change this class in this PR anyways?

Reply:
Ahh this is a side effect of a refactor to support the Flash transactions (before I changed approach and decided we should just map them into a dummy receipt) - which I then mostly reverted, but this remained!

I didn't realise DescribedTransactionHash was intended just for logging hah - but this is private to this class so I'm not particularly worried :).

I can rename this parameter to `described_hash` if that makes it less scary? 👍

Reply:
> I can rename this parameter

Either this (accompanied by update to the rustdoc of DescribedTransactionHash)...
...OR - if the changes to this file really turned out to not be needed - then I'd be even more happy if you simply reverted the file.